src/services/pallets/PalletsErrorsService.ts

```diff
@@ -0,0 +1,298 @@
+// Copyright 2017-2022 Parity Technologies (UK) Ltd.
+// This file is part of Substrate API Sidecar.
+//
+// Substrate API Sidecar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import { ApiDecoration, ModuleErrors } from '@polkadot/api/types';
+import { Text, Vec } from '@polkadot/types';
+import {
+	BlockHash,
+	ErrorMetadataLatest,
+	MetadataV14,
+	PalletErrorMetadataV14,
+	PalletMetadataV14,
+} from '@polkadot/types/interfaces';
+import { IsError } from '@polkadot/types/metadata/decorate/types';
+import { stringCamelCase } from '@polkadot/util';
+import { BadRequest, InternalServerError } from 'http-errors';
+import {
+	IPalletErrors,
+	IPalletErrorsItem,
+	ISanitizedErrorItemMetadata,
+} from 'src/types/responses';
+
+import { sanitizeNumbers } from '../../sanitize/sanitizeNumbers';
+import { AbstractService } from '../AbstractService';
+
+interface IFetchPalletArgs {
+	hash: BlockHash;
+	palletId: string;
+}
+
+interface IFetchErrorItemArgs extends IFetchPalletArgs {
+	errorItemId: string;
+	metadata: boolean;
+}
+
+export class PalletsErrorService extends AbstractService {
+	async fetchErrorItem(
+		historicApi: ApiDecoration<'promise'>,
+		{ hash, palletId, errorItemId, metadata }: IFetchErrorItemArgs
+	): Promise<IPalletErrorsItem> {
+		const palletMetadata = historicApi.registry.metadata;
+
+		const [palletMeta, palletMetaIdx] = this.findPalletMeta(
+			palletMetadata,
+			historicApi,
+			palletId
+		);
+
+		const palletName = stringCamelCase(palletMeta.name);
+		const errors = historicApi.errors[palletName];
+
+		// Even if `errorItemMeta` is not used, we call this function to ensure it exists. The side effects
+		// of the error item not existing are that `findErrorItemMeta` will throw.
+		const errorItemMetadata = this.findErrorItemMeta(
+			palletMeta,
+			errorItemId,
+			errors
+		);
+
+		let normalizedErrorItemMeta;
+		if (metadata) {
+			normalizedErrorItemMeta = this.normalizeErrorItemMeta(
+				(errorItemMetadata[1] as unknown as IsError).meta
```

Comment:
Whats the case for type casting as unknown here? Is the `errorItemMetadata[1]` recognized as just `Codec` so we have to specify the type?

Reply:
good catch! I think this casting to unknown can be removed

Reply:
Awesome